### PR TITLE
New serverless pattern : Lambda event generation, their distribution via EventBridge rules to rate controlled processing using SQS+Lambda

### PR DESCRIPTION
**Pull request by anaghlal**
*Issue #, if available:*

*Description of changes:*
Lambda event generation, their distribution via EventBridge rules to rate controlled processing using SQS+Lambda

The pattern demonstrates creating and sending events to EventBridge, routing them to targets based on message contents to SQS targets consumed by Lambdas.
 "language": "Python"
"framework": "CDK"

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by anaghlal** [comment]
I have submitted a change with the title and description meeting the max length limits.  How do I retrigger the bot?

File: lambda-eventbridge-sqs-lambda-cdk-python/app.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            self,
+            "CfnRole",
+            assumed_by=iam.ServicePrincipal("events.amazonaws.com"),
+        )
+        lambda_cfn_role.add_managed_policy(
+            iam.ManagedPolicy.from_aws_managed_policy_name("AWSLambdaExecute")
+        )
+
+        # lambda function for processing the incoming request from S3 bucket
+        lambda_function = lambda_.Function(
```
Comment (ellisms) [inline lambda-eventbridge-sqs-lambda-cdk-python/app.py:39]: I would explicitly define the function log groups and deletion policy so they are cleaned up during cdk destroy.

**Review by ellisms — CHANGES_REQUESTED** [review]
Minor edits/comments to review before merging.

File: lambda-eventbridge-sqs-lambda-cdk-python/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+pip install -r requirements.txt
+```
+
+1. From the command line, configure AWS CDK:
+   ```bash
+   cdk bootstrap ACCOUNT-NUMBER/REGION # e.g.
+   cdk bootstrap 1111111111/us-east-1
+   cdk bootstrap --profile test 1111111111/us-east-1
+   ```
+1. From the command line, use AWS CDK to deploy the AWS resources for the pattern as specified in the `stack/s3-eventbridge-lambda-stack.ts`
```
[the remark this answers is not in the document]
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/README.md]: fixed

File: lambda-eventbridge-sqs-lambda-cdk-python/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+   ```bash
+   cdk deploy
+   ```
+
+
+## How it works
```
[the remark this answers is not in the document]
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/README.md:50]: rephrased

File: lambda-eventbridge-sqs-lambda-cdk-python/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- Go to the Lambda ARN printed as part of the CDK output. Do a Test invocation of the Lambda from the console.
+- This will send three events to EventBridge and each will be routed to either the email/sftp/3papi SQS queue.
+- Finally each of the processing Lambdas get triggered by the message on their respective queues.
+
+
+## Testing
+
+1. Go to the Lambda ARN printed as part of the CDK output. Do a Test invocation of the Lambda from the console.
+1. This will send three events to EventBridge and each will be routed to either the email/sftp/3papi SQS queue and eventually to the processor Lambdas.
+
```
[the remark this answers is not in the document]
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/README.md:63]: resolved cloudwatch logs for each lambda

File: lambda-eventbridge-sqs-lambda-cdk-python/app.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        lambda_cfn_role = iam.Role(
+            self,
+            "CfnRole",
+            assumed_by=iam.ServicePrincipal("events.amazonaws.com"),
+        )
+        lambda_cfn_role.add_managed_policy(
+            iam.ManagedPolicy.from_aws_managed_policy_name("AWSLambdaExecute")
+        )
+
+        # lambda function for processing the incoming request from S3 bucket
```
[the remark this answers is not in the document]
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/app.py]: cleaned up

File: lambda-eventbridge-sqs-lambda-cdk-python/app.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        
+        
+        # lambda version
+        version = lambda_function.current_version
+
+        # lambda policy
+        lambda_function.add_to_role_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "events:CreateEventBus",
```
[the remark this answers is not in the document]
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/app.py]: removed extra access

File: lambda-eventbridge-sqs-lambda-cdk-python/app.py (hunk 1 above)
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/app.py:42]: done

File: lambda-eventbridge-sqs-lambda-cdk-python/README.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
+# Event generation, their distribution via Amazon EventBridge rules to rate controlled processing using Amazon SQS and AWS Lambda
+
+This pattern contains a sample AWS Cloud Development Kit (AWS CDK) template for creating a Lambda function that posts AWS EventBridge events to the default domain bus. This CDK template defines EventBridge rules to distribute events based on rules to SQS queues also created in the CK template.  This CDK template also deploys a AWS Lambda functions linked to each queue for event-type specific processing.
+
+Learn more about this pattern at Serverless Land Patterns:: https://serverlessland.com/patterns/s3-eventbridge-lambda
```
[the remark this answers is not in the document]
Comment (anaghlal) [reply lambda-eventbridge-sqs-lambda-cdk-python/README.md]: done

**Comment by ellisms** [comment]
Changes look good, thanks! 

One last question - since the pattern is focussed on rate limiting, can you feature the [Lambda event source max concurrency ](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-max-concurrency)feature in the example? I think that would tie the usage of SQS as the buffer to rate limiting.

**Review by ellisms — APPROVED** [review]

**Comment by bls20AWS** [comment]
live: https://serverlessland.com/patterns/lambda-eb-sqs-lambda

**Pull request merged**